src/machine/machine_tkey.go

```diff
@@ -0,0 +1,227 @@
+//go:build tkey
+
+package machine
+
+import (
+	"device/tkey"
+	"strconv"
+)
+
+const deviceName = "TKey"
+
+// GPIO pins modes are only here to match the Pin interface.
+// The actual configuration is fixed in the hardware.
+const (
+	PinOutput PinMode = iota
+	PinInput
+	PinInputPullup
+	PinInputPulldown
+)
+
+const (
+	LED_BLUE  = Pin(tkey.TK1_MMIO_TK1_LED_B_BIT)
+	LED_GREEN = Pin(tkey.TK1_MMIO_TK1_LED_G_BIT)
+	LED_RED   = Pin(tkey.TK1_MMIO_TK1_LED_R_BIT)
+
+	LED = LED_GREEN
+
+	TKEY_TOUCH = Pin(3) // 3 is unused, but we need a value here to match the Pin interface.
+	BUTTON     = TKEY_TOUCH
+
+	GPIO1 = Pin(tkey.TK1_MMIO_TK1_GPIO1_BIT + 8)
+	GPIO2 = Pin(tkey.TK1_MMIO_TK1_GPIO2_BIT + 8)
+	GPIO3 = Pin(tkey.TK1_MMIO_TK1_GPIO3_BIT + 8)
+	GPIO4 = Pin(tkey.TK1_MMIO_TK1_GPIO4_BIT + 8)
+)
+
+var touchConfig, gpio1Config, gpio2Config PinConfig
+
+// No config needed for TKey, just to match the Pin interface.
+func (p Pin) Configure(config PinConfig) {
+	switch p {
+	case BUTTON:
+		touchConfig = config
+
+		// Clear any pending touch events.
+		tkey.TOUCH.STATUS.Set(0)
+	case GPIO1:
+		gpio1Config = config
+	case GPIO2:
+		gpio2Config = config
+	}
+}
+
+// Set pin to high or low.
+func (p Pin) Set(high bool) {
+	switch p {
+	case LED_BLUE, LED_GREEN, LED_RED:
+		if high {
+			tkey.TK1.LED.SetBits(1 << uint(p))
+		} else {
+			tkey.TK1.LED.ClearBits(1 << uint(p))
+		}
+	case GPIO3, GPIO4:
+		if high {
+			tkey.TK1.GPIO.SetBits(1 << uint(p-8))
+		} else {
+			tkey.TK1.GPIO.ClearBits(1 << uint(p-8))
+		}
+	}
+}
+
+// Get returns the current value of a pin.
+func (p Pin) Get() bool {
+	pushed := false
+	mode := PinInput
+
+	switch p {
+	case BUTTON:
+		mode = touchConfig.Mode
+		if tkey.TOUCH.STATUS.HasBits(1) {
+			tkey.TOUCH.STATUS.Set(0)
+			pushed = true
+		}
+	case GPIO1:
+		mode = gpio1Config.Mode
+		if tkey.TK1.GPIO.HasBits(1 << uint(p-8)) {
+			pushed = true
+		}
+	case GPIO2:
+		mode = gpio2Config.Mode
+		if tkey.TK1.GPIO.HasBits(1 << uint(p-8)) {
+			pushed = true
+		}
+	}
+
+	switch mode {
+	case PinInputPullup:
+		return !pushed
+	case PinInput, PinInputPulldown:
+		return pushed
+	}
+
+	return false
+}
+
+type UART struct {
+	Bus *tkey.UART_Type
+}
+
+var (
+	DefaultUART = UART0
+	UART0       = &_UART0
+	_UART0      = UART{Bus: tkey.UART}
+)
+
+// Thw TKey UART is fixed at 62500 baud, 8N1.
```

Comment:
Typo

```suggestion
// The TKey UART is fixed at 62500 baud, 8N1.
```

Also, you could consider returning an error if any other configuration than this one is specified by the user.

Reply:
Good idea, done.